libbeat/outputs/kafka/config_test.go

```diff
@@ -104,6 +114,76 @@ func TestConfigInvalid(t *testing.T) {
 	}
 }
 
+func TestConfigUnderElasticAgent(t *testing.T) {
+	oldUnderAgent := management.UnderAgent()
+	t.Cleanup(func() {
+		// Restore the previous value
+		management.SetUnderAgent(oldUnderAgent)
+	})
+
+	management.SetUnderAgent(true)
+
+	tests := []struct {
+		name        string
+		cfg         mapstr.M
+		expectError bool
+	}{
+		{
+			name: "topic with all valid characters",
+			cfg: mapstr.M{
+				"topic": "abcdefghijklmnopqrstuvxz-ABCDEFGHIJKLMNOPQRSTUVXZ_01234567890.",
+			},
+		},
+		{
+			name: "topics is provided",
+			cfg: mapstr.M{
+				"topics": []string{"foo", "bar"},
+			},
+			expectError: true,
+		},
+		{
+			name: "topic cannot contain invalid characters",
+			cfg: mapstr.M{
+				"topic": "foo bar",
+			},
+			expectError: true,
+		},
+		{
+			name: "topic with invalid characters",
+			cfg: mapstr.M{
+				"topic": "foo + bar",
```

Comment:
I would specifically add a test case with the `%{}` syntax to prove we detect it as invalid.

Reply:
It might be a good idea to have a test to prove this isn't invalid and is successfully  substituted when not running under agent, but that test might already exist.

Reply:
I've added the test case on e6280db463

For standalone Beats the test case already exist: https://github.com/elastic/beats/blob/9c9ae3568309bb4c715fe1c25834eafd06f0f82a/libbeat/outputs/kafka/config_test.go#L163-L176